### PR TITLE
Better file Creation/Renaming - WIP

### PR DESCRIPTION
**Pull request by DomaDude123**

### CODE REVIEW

File: src/client/App.tsx [hunk 1]
```diff
@@ -186,33 +169,43 @@ function App() {
...
-  usePersistUsername(username);
-
   // Cache of CodeMirror editors by file id.
   const editorCache: EditorCache = useEditorCache();
 
+  // A function that will focus the editor.
+  const focusEditor = useCallback(
+    (fileId: FileId) => {
+      // TODO figure out a way to do this without the timeout
+      setTimeout(() => {
```
Comment (curran) [inline src/client/App.tsx:186]: It looks like you started from an older version of the code, and this PR is bringing back these old bits.

**Review by curran — CHANGES_REQUESTED** [review]
It looks like there's some old stuff in this PR that reverts things to an older version.

It might make sense to start fresh, branching from `main`, then add the new modal to the clean slate.

Will do local testing later to help identify the bug.

Thanks for this!

File: src/client/VZSidebar/index.tsx [hunk 2]
```diff
@@ -142,6 +122,13 @@ export const VZSidebar = ({
           Settings
         </div>
       )}
+      <CreateFileModal
```
Comment (curran) [inline src/client/VZSidebar/index.tsx:125]: Please run `npm run prettier` to address formatting issues. Thanks!

File: src/client/useFileCRUD.ts [hunk 3]
```diff
@@ -1,42 +1,33 @@
...
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const createFile = useCallback((fileName) => {
     // TODO use Bootstrap modals not native prompts
     // See https://github.com/vizhub-core/vzcode/issues/252
-    const name = prompt('Enter new file name');
-    if (name) {
+    //const name = prompt('Enter new file name');
+    
+    console.log("Here's filename in createFile- ", fileName)
```
Comment (curran) [inline src/client/useFileCRUD.ts:22]:
```suggestion
```

File: src/client/useFileCRUD.ts [hunk 4]
```diff
@@ -94,10 +85,30 @@ export const useFileCRUD = ({
     [submitOperation, closeTabs],
   );
 
+  // Generic delete function that calls either deleteFile or deleteDirectory
+  // TODO consider if there is a cleaner way to do this
+  //  - Ideally we would have a separate function for deleting files and directories
+  //  - When we click the delete button, we should be able to tell if it is a file or directory
+  const handleDeleteClick = useCallback(
```
Comment (curran) [inline src/client/useFileCRUD.ts:92]: Yeah this is old, it was replaced by new stuff in a refactoring.

**Comment by DomaDude123** [comment]
> It looks like there's some old stuff in this PR that reverts things to an older version.
> 
> It might make sense to start fresh, branching from `main`, then add the new modal to the clean slate.
> 
> Will do local testing later to help identify the bug.
> 
> Thanks for this!

I'm going to create a new branch and implement the Modal there. Will try to find that bug along the way and create a new PR when ready. Thanks!

**Pull request closed without merging**